### ISSUE TITLE
Parameter sharing not working? (BrainScript)

### ISSUE THREAD

**Issue by vermorel** [Microsoft/CNTK#2572]
I am facing a problem with BrainScript, and I am wondering if parameter sharing works as expected. Here is the relevant snippet of code:

        MyLayer { } = {
            apply(x) = {
                A = DenseLayer {128, activation = ReLU}(x) 
            }.A
        }.apply
     
       MX = MyLayer {} # MX parameters should be shared

       AlphaX = Splice(array [0.99] ( i => {  
            A = MX(DelayLayer { T = i }( SomeInput )) )
        }.A ))

The snipped above ends-up generating tons of parameters, not seemingly applying any parameter sharing logic. Normally,  I would expect `MX` parameters to be shared across every array item. Is there something wrong with BrainScript? Is my understanding of the parameter sharing incorrect?

Thanks in advance.

**Comment by kaituoxu** [comment]
@vermorel  
Have you solved this problem?
I met a similar problem.
It seems due to the array operation.

**Comment by vermorel** [comment]
@kaituoxu No, I never did solve this problem. In the end, I am avoiding arrays pretty much everywhere, sometimes using [initFromFilePath](https://docs.microsoft.com/en-us/cognitive-toolkit/Parameters-And-Constants) as a workaround.

**Comment by kaituoxu** [comment]
@vermorel Thanks for your response. 
I met a similar problem at https://github.com/Microsoft/CNTK/issues/3025, can you help me on this issue? What should I use to replace array?
Thank you very much.
